no_panic and const

Does `no_panic` work with const functions? If not, It should be mentioned in the documentation, I think.

```rs
error[E0015]: cannot call non-const closure in constant functions
   --> crates\bevy_math\src\rotation2d.rs:224:5
    |
224 |     #[no_panic]
    |     ^^^^^^^^^^^
    |
    = note: closures need an RFC before allowed to be called in constant functions
    = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants
    = note: this error originates in the attribute macro `no_panic` (in Nightly builds, run with -Z macro-backtrace for more info)
```
```rs
    #[no_panic]
    pub const fn sin_cos(self) -> (f32, f32) {
        (self.sin, self.cos)
    }
```

--- a/README.md
+++ b/README.md
@@ -75,6 +75,9 @@ provides the name of the offending function.
   detected. After confirming absence of panics, you can of course still ship
   your software as a `panic = "abort"` build.
 
+- Const functions are not supported. The attribute will fail to compile if
+  placed on a `const fn`.
+
 If you find that code requires optimization to pass `#[no_panic]`, either make
 no-panic an optional dependency that you only enable in release builds, or add a
 section like the following to your Cargo.toml or .cargo/config.toml to enable

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -84,6 +84,9 @@
 //!   be detected. After confirming absence of panics, you can of course still
 //!   ship your software as a `panic = "abort"` build.
 //!
+//! - Const functions are not supported. The attribute will fail to compile if
+//!   placed on a `const fn`.
+//!
 //! If you find that code requires optimization to pass `#[no_panic]`, either
 //! make no-panic an optional dependency that you only enable in release builds,
 //! or add a section like the following to your Cargo.toml or .cargo/config.toml

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -168,6 +168,12 @@ pub fn no_panic(args: TokenStream, input: TokenStream) -> TokenStream {
 fn parse(args: TokenStream2, input: TokenStream2) -> Result<ItemFn> {
     let function: ItemFn = syn::parse2(input)?;
     let _: Nothing = syn::parse2::<Nothing>(args)?;
+    if function.sig.constness.is_some() {
+        return Err(Error::new(
+            Span::call_site(),
+            "no_panic attribute on const fn is not supported",
+        ));
+    }
     if function.sig.asyncness.is_some() {
         return Err(Error::new(
             Span::call_site(),

--- a/tests/ui/const-fn.rs
+++ b/tests/ui/const-fn.rs
@@ -1,0 +1,6 @@
+use no_panic::no_panic;
+
+#[no_panic]
+const fn f() {}
+
+fn main() {}

--- a/tests/ui/const-fn.stderr
+++ b/tests/ui/const-fn.stderr
@@ -1,0 +1,9 @@
+error[E0015]: cannot call non-const closure in constant functions
+ --> tests/ui/const-fn.rs:3:1
+  |
+3 | #[no_panic]
+  | ^^^^^^^^^^^
+  |
+  = note: closures need an RFC before allowed to be called in constant functions
+  = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants
+  = note: this error originates in the attribute macro `no_panic` (in Nightly builds, run with -Z macro-backtrace for more info)

--- a/tests/ui/const-fn.stderr
+++ b/tests/ui/const-fn.stderr
@@ -1,9 +1,7 @@
-error[E0015]: cannot call non-const closure in constant functions
+error: no_panic attribute on const fn is not supported
  --> tests/ui/const-fn.rs:3:1
   |
 3 | #[no_panic]
   | ^^^^^^^^^^^
   |
-  = note: closures need an RFC before allowed to be called in constant functions
-  = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants
   = note: this error originates in the attribute macro `no_panic` (in Nightly builds, run with -Z macro-backtrace for more info)

